cmd/meroxa/turbine/python/deploy.go

```diff
@@ -2,106 +2,110 @@ package turbinepy
 
 import (
 	"context"
-	"encoding/json"
-	"errors"
-	"fmt"
-	"os"
-	"os/exec"
-	"regexp"
-	"strconv"
-	"strings"
-
-	"github.com/meroxa/cli/cmd/meroxa/global"
+	"time"
+
 	utils "github.com/meroxa/cli/cmd/meroxa/turbine"
-	"github.com/meroxa/cli/log"
+	"github.com/meroxa/cli/cmd/meroxa/turbine/python/internal"
+	pb "github.com/meroxa/turbine-core/lib/go/github.com/meroxa/turbine/core"
+	"github.com/meroxa/turbine-core/pkg/client"
+	"google.golang.org/protobuf/types/known/emptypb"
 )
 
 // NeedsToBuild determines if the app has functions or not.
-func (t *turbinePyCLI) NeedsToBuild(_ context.Context) (bool, error) {
-	cmd := exec.Command("turbine-py", "hasFunctions", t.appPath)
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		err = fmt.Errorf(
-			"unable to determine if the Meroxa Application at %s has a Process; %s",
-			t.appPath,
-			string(output))
-		return false, err
-	}
 
-	isNeeded, err := utils.GetTurbineResponseFromOutput(string(output))
+func (t *turbinePyCLI) NeedsToBuild(ctx context.Context) (bool, error) {
+	ctx, cancel := context.WithTimeout(ctx, time.Second)
+	defer cancel()
+	resp, err := t.bc.HasFunctions(ctx, &emptypb.Empty{})
 	if err != nil {
-		err := fmt.Errorf(
-			"unable to determine if the Meroxa Application at %s has a Process; %s",
-			t.appPath,
-			string(output))
 		return false, err
 	}
-	return strconv.ParseBool(isNeeded)
+
+	return resp.Value, nil
 }
 
-// Deploy creates Application entities.
-func (t *turbinePyCLI) GetDeploymentSpec(ctx context.Context, imageName, appName, gitSha, specVersion, accountUUID string) (string, error) {
-	var (
-		output         string
-		deploymentSpec string
+func (t *turbinePyCLI) CreateDockerfile(ctx context.Context, _ string) (string, error) {
+	cmd := internal.NewTurbineCmd(t.appPath,
+		internal.TurbineCommandBuild,
+		map[string]string{},
+		t.appPath,
 	)
-	args := []string{"clideploy", t.appPath, imageName, appName, gitSha}
+	return t.appPath, utils.RunCMD(ctx, t.logger, cmd)
+}
 
-	if specVersion != "" {
-		args = append(args, specVersion)
-	}
+func (t *turbinePyCLI) CleanUpBuild(_ context.Context) {
+	utils.CleanupDockerfile(t.logger, t.appPath)
+}
 
-	cmd := exec.CommandContext(ctx, "turbine-py", args...)
+func (t *turbinePyCLI) SetupForDeploy(ctx context.Context, gitSha string) (func(), error) {
+	go t.builder.Run(ctx)
```

Comment:
Maybe we could use this one instead to randomize its port (looks like we've got this one already randomized here https://github.com/meroxa/cli/pull/725/files#diff-422ee0039de1ccac947bc720de54bf7c703245ffc16d95c1c6bc5cd02025cd35R24):

```suggestion
	go t.builder.RunAddr(ctx, t.grpcListenAddress)
```

Reply:
Good catch! Thanks :D